src/testrender/shading.cpp

```diff
@@ -491,7 +528,22 @@ struct OrenNayar final : public BSDF, OrenNayarParams {
     }
 
 private:
-    float A, B;
+    static constexpr float constant1_FON = float(0.5 - 2.0 / (3.0 * M_PI));
+    static constexpr float constant2_FON = float(2.0 / 3.0
+                                                 - 28.0 / (15.0 * M_PI));
+
+    float E_FON_analytic(float mu) const
+    {
+        float AF = 1.0f
+                   / (1.0f
+                      + constant1_FON * sigma);  // Fujii model A coefficient
+        float BF = sigma * AF;                   // Fujii model B coefficient
+        float Si = sqrtf(std::max(0.0f, 1.0f - mu * mu));
+        float G  = Si * (OIIO::fast_acos(mu) - Si * mu)
```

Comment:
Why is the fast_acos required here?
If you look at my version: https://www.desmos.com/calculator/extaj6cnzr the E3(x) function I define can take mu directly as a parameter and provides a very high quality polynomial fit to give the same result.
It's quite a lot faster numerically. Curious if you would be interested in testing the performance. It also doesn't require special handling as it has no issues at either 0 or 1. It should provide a significant performance improvement over calling acos.

Reply:
i.e.
![image](https://github.com/AcademySoftwareFoundation/OpenShadingLanguage/assets/1401669/d30e0b46-efd8-478f-9b52-e975949eeee2)
![image](https://github.com/AcademySoftwareFoundation/OpenShadingLanguage/assets/1401669/376de29b-a408-485b-9233-f7323b243c1b)

Reply:
@etheory Thanks for your notes, that's helpful. Very interesting that you also found a similar functional form.

We also did a fit to that G function, which has maximum fraction error of 6e-4 over all directions. We can compare to your formulation as well.

If it turns out to be better, we'd be happy to credit you in the write-up, if you don't mind sharing your real name.

Reply:
Sure thing. My name is Luke Emrose. Engineering Manager for Rendering at Animal Logic. More than happy for you to use it if it's useful.

Reply:
Thanks Luke! Obviously if we find something in your notes that we use in the write-up, we will let you know and credit you in the acknowledgements.

Reply:
Yeah I went for the vanilla analytic formula for now to have as a reference. I fully expect swapping it for one of the approximations in a second pass.